.github/workflows/CI_Pipeline.yml

```diff
@@ -0,0 +1,39 @@
+name: CI Pipeline
+
+on:
+  push:
+    branches: '**'
+
+jobs:
+  build:
+    runs-on: windows-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up JDK 21
+      uses: actions/setup-java@v4
+      with:
+        java-version: '21'
+        distribution: 'temurin'
+        cache: gradle
+
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@af1da67850ed9a4cedd57bfd976089dd991e2582
+      with:
+        gradle-version: '8.9'
+
+    - name: Cache Gradle packages
```

Comment:
Incase the dependencies are updated or newly added, how will they be handled? Will the CI pipeline use the old cache or generate a new cache?